### PR TITLE
V2 : Adding 3D hull functionality

### LINKED ISSUE #226: implement 3d convex hull
I needed hull to do OpenSCAD's full 3D convex hull so I integrated [quickhull3d](https://github.com/maurizzzio/quickhull3d) into my [fork of OpenJSCAD](https://github.com/AllwineDesigns/OpenJSCAD.org/commit/2757a87c4baca2348cdb7112421aefdea9e03946). I'm jammed for time right now, but I'll try to loop back and get a decent pull request together. 

Here's briefly what I did:

- I used browserify to package up quickhull3d and included it as hull3d.js
- I added hull3d.js to the list of included libraries when running the jscad worker
- I modified the hull function to check if any arguments aren't CAG objects and, if so, do a 3D hull operation

### PR DESCRIPTION
**Pull request by z3dev**
This pull requests adds hull operations for 3D geometries. The 3D hull implementation was taken and adapted to JSCAD standards from the quickhull library.

- [X] code complete
- [X] tests complete, 96.45% (2D hull 97.06%, 3D hull 86.19%)
- [X] review complete

Fixes #226 
Fixes CSG # 113

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?
* [X] Does your submission pass tests?

### CODE REVIEW

**Comment by z3dev** [comment]
This is working very nicely.

@kaosat-dev please review.

File: packages/modeling/src/operations/hulls/quickhull/Face.js [hunk 1]
```diff
@@ -0,0 +1,327 @@
+const add = require('../../../math/vec3/add')
```
Comment (kaosat-dev) [inline packages/modeling/src/operations/hulls/quickhull/Face.js]: Nice work adapting the code of quickhull3D to our stuff !
I think that for each of these quickhull files we should be very clear that they are not based on our code but are a modified versions of https://www.npmjs.com/package/quickhull3d (licence & attribution of code origin are very important :)

File: packages/modeling/src/operations/hulls/quickhull/point-line-distance.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * Copyright (c) 2015 Mauricio Poppe
+ * Licensed under the MIT license.
+ */
+
+const cross = require('../../../math/vec3/cross')
+const subtract = require('../../../math/vec3/subtract')
+const squaredLength = require('../../../math/vec3/squaredLength')
+
+const distanceSquared = (p, a, b) => {
```
Comment (kaosat-dev) [inline packages/modeling/src/operations/hulls/quickhull/point-line-distance.js]: should we perhaps move this one to maths at some point ? I can imagine it being useful

File: packages/modeling/src/operations/hulls/quickhull/Face.js (hunk 1 above)
Comment (z3dev) [reply packages/modeling/src/operations/hulls/quickhull/Face.js]: @kaosat-dev please review the files again. i added copyright and information to each of the files in the quickhull library.

Comment (kaosat-dev) [reply packages/modeling/src/operations/hulls/quickhull/Face.js]: @z3dev : thanks for updating !
One last thing is having that information in the README as well and it is all golden ! :)

Comment (z3dev) [reply packages/modeling/src/operations/hulls/quickhull/Face.js]: I added copyright information to README.md for the quickhull library.

I was thinking... shouldn't we add a copyright for the gl-matrix library?

Comment (kaosat-dev) [reply packages/modeling/src/operations/hulls/quickhull/Face.js]: hmm that is a good tricky question
we are not using the code as such and we have diverged a fair bit, but perhaps it is both more nice, correct and open to clearly put 'the maths library and modularity is based on the gl-matrix and its split up versions'
I will add this to the list of needed tweaks before V2 release, and merge this in the meantime

**Pull request merged**